wp-modules/app/js/src/hooks/useEnvironment.ts

```diff
@@ -0,0 +1,25 @@
+import { useState } from '@wordpress/element';
+import { patternManager } from '../globals';
+import getHeaders from '../utils/getHeaders';
+
+export default function useEnvironment( initialEnvironmentNotice: boolean ) {
+	const [ displayNotice, setDisplayNotice ] = useState(
+		initialEnvironmentNotice
+	);
+
+	function updateDismissedSites() {
+		setDisplayNotice( false );
+		return fetch(
```

Comment:
It's nice how this `fetch()` is orthogonal to the `displayNotice` state.

There's no need to wait for the `fetch()` call to complete to update the `displayNotice` state.

Reply:
Learned that one from you! Optimistic UI for the win 😊

Reply:
Haha, thanks!